### PR TITLE
Dockerfile changes to enable correct collectd functioning under alpine

### PR DESCRIPTION
**Pull request by NeilCondon**
## Proposed changes

Include packages and Dockerfile changes necessary to make collectd push metrics correctly (refer to #37).
Update the docker-alpine project to build thin/edge 0.7.7 by default.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New example (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I have added necessary documentation (if appropriate)

## Further comments

My mistake not to sign the commits individually - sorry.

### CODE REVIEW

File: docker-alpine/Dockerfile [hunk 1]
```diff
@@ -22,16 +22,19 @@ VOLUME ["/sys/fs/cgroup"]
...
-RUN chmod 1777 /run/lock/
+    && apk --no-cache add mosquitto curl collectd collectd-mqtt sudo openrc
 
 # Add user and groups
 RUN addgroup -S tedge
 RUN adduser -g "" -H -D tedge -G tedge
 
+# Create useful directories
+RUN mkdir /run/lock && \
+    mkdir /var/tedge && \
```
Comment (rina23q) [inline docker-alpine/Dockerfile:33]: `/var/tedge` should be created by `RUN sudo -u tedge tedge_agent --init`, so it's not necessary.

File: docker-alpine/Dockerfile [hunk 2]
```diff
@@ -22,16 +22,19 @@ VOLUME ["/sys/fs/cgroup"]
...
+    mkdir /etc/collectd/collectd.conf.d && \
+    chmod 1777 /run/lock/ && \
+    chown tedge:tedge /var/tedge
```
Comment (rina23q) [inline docker-alpine/Dockerfile:36]: Also, `RUN sudo -u tedge tedge_agent --init` creates `/var/tedge` by the owner `tedge:tedge`. This line can be also omitted.

**Review by rina23q — APPROVED** [review]
I didn't check if the collectd part works on my machine, but overall it looks good.

**Comment by reubenmiller** [comment]
Closing as the docker-alpine example is being deprecated as the [tedge-demo-container](https://github.com/thin-edge/tedge-demo-container) includes a more complete example showing how to use thin-edge.io in a container environment (both multi-service container and single process container setup)

**Pull request closed without merging**